src/Plugins/PluginInfo.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Pantheon\Terminus\Plugins;
+
+use Consolidation\AnnotatedCommand\CommandFileDiscovery;
+use Pantheon\Terminus\Exceptions\TerminusException;
+
+class PluginInfo
+{
+    /**
+     * @var string
+     */
+    protected $plugin_dir;
+
+    /**
+     * @var null|array
+     */
+    protected $info = null;
+
+    /**
+     * PluginInfo constructor.
+     * @param $plugin_dir
+     */
+    public function __construct($plugin_dir)
+    {
+        $this->plugin_dir = $plugin_dir;
+        $this->info = $this->parsePluginInfo();
+    }
+
+    /**
+     * Get the info array for the plugin.
+     *
+     * @return array|null|string
+     */
+    public function getInfo() {
+        return $this->info;
+    }
+
+    /**
+     * Get all of the commands and hooks in the plugin.
+     *
+     * @return array
+     */
+    public function getCommandsAndHooks() {
+        $path = $this->getCommandFileDirectory();
+        $namespace = $this->getCommandNamespace();
+        $discovery = new CommandFileDiscovery();
+        $discovery->setSearchPattern('*Command.php')->setSearchLocations([])->setSearchDepth(4);
```

Comment:
Wouldn't it be better to set some search locations, and limit the search depth, so that you do not need to search the entire plugin source for commandfiles?